pkg/capture/capture.go

```diff
@@ -256,26 +235,48 @@ func NewCapture(ctx context.Context, iface string, config config.CaptureConfig)
 	return &Capture{
 		iface:         iface,
 		mutex:         sync.Mutex{},
+		stateMutex:    sync.RWMutex{},
 		config:        config,
 		cmdChan:       make(chan captureCommand),
 		captureErrors: make(chan error),
 		lastRotationStats: capturetypes.PacketStats{
 			CaptureStats: &capturetypes.CaptureStats{},
 		},
 		rotationState: newRotationState(),
+		closeState:    make(chan struct{}, 1),
 		flowLog:       capturetypes.NewFlowLog(),
 		errMap:        make(map[string]int),
 		ctx:           capCtx,
+		sourceInitFn:  defaultSourceInitFn,
 	}
 }
 
+// Iface returns the interface name of the Capture
+func (c *Capture) Iface() string {
+	return c.iface
+}
+
+// SetSourceInitFn sets a custom function used to initialize a new capture
+func (c *Capture) SetSourceInitFn(fn sourceInitFn) *Capture {
+	c.sourceInitFn = fn
+	return c
+}
+
 // stateFn enables the implementation of the state machine
 type stateFn func(*Capture) stateFn
 
+// sourceInitFn denotes the function used to initialize a capture source,
+// providing the ability to override the default behavior, e.g. in mock tests
+type sourceInitFn func(*Capture) (capture.Source, error)
```

Comment:
Here be magic. In order to be able to inject mock sources from `slimcap` I've allowed sources to override the default init function (so that e.g. mock sources can be initialized together with external tracking mechanisms and the likes). Also check out `defaultSourceInitFn` further up.